go/libraries/doltcore/merge/merge_prolly_rows.go

```diff
@@ -1532,40 +1544,131 @@ func (m *valueMerger) tryMerge(left, right, base val.Tuple) (val.Tuple, bool) {
 
 // processColumn returns the merged value of column |i| of the merged schema,
 // based on the |left|, |right|, and |base| schema.
-func (m *valueMerger) processColumn(i int, left, right, base val.Tuple) ([]byte, bool) {
+func (m *valueMerger) processColumn(ctx context.Context, i int, left, right, base val.Tuple) ([]byte, bool) {
 	// missing columns are coerced into NULL column values
-	var leftCol []byte
-	if l := m.leftMapping[i]; l != -1 {
-		leftCol = left.GetField(l)
+
+	var baseCol []byte
+	var baseColIdx = -1
+	var baseColExists = false
+	if base != nil {
+		baseCol, baseColIdx, baseColExists = getColumn(&base, &m.baseMapping, i)
 	}
-	var rightCol []byte
-	if r := m.rightMapping[i]; r != -1 {
-		rightCol = right.GetField(r)
+	leftCol, leftColIdx, leftColExists := getColumn(&left, &m.leftMapping, i)
+	rightCol, rightColIdx, rightColExists := getColumn(&right, &m.rightMapping, i)
```

Comment:
Can `left` and `right` be nil, too? We're checking `base` for nil, but seems like `left` and `right` could also be nil. Would be nice if `getColumn` could handle that just so that the variable initialize code doesn't have to take up 5 lines for each side of the three way merge.